File: benches/list.rs
use criterion::*;

fn list(c: &mut Criterion) {
    let mut group = c.benchmark_group("list");

    #[cfg(target_os = "windows")]
    group.bench_function("amazon::games", |b| {
        b.iter(|| {
            let games = game_scanner::amazon::games().unwrap();
            assert_ne!(games.len(), 0);
        })
    });

    #[cfg(not(target_os = "linux"))]
    group.bench_function("blizzard::games", |b| {
        b.iter(|| {
            let games = game_scanner::blizzard::games().unwrap();
            assert_ne!(games.len(), 0);
        })
    });

    #[cfg(not(target_os = "linux"))]
    group.bench_function("epicgames::games", |b| {
        b.iter(|| {
            let games = game_scanner::epicgames::games().unwrap();
            assert_ne!(games.len(), 0);
        })
    });

    #[cfg(not(target_os = "linux"))]
    group.bench_function("gog::games", |b| {
        b.iter(|| {
            let games = game_scanner::gog::games().unwrap();
            assert_ne!(games.len(), 0);
        })
    });

    #[cfg(not(target_os = "linux"))]
    group.bench_function("origin::games", |b| {
        b.iter(|| {
            let games = game_scanner::origin::games().unwrap();
            assert_ne!(games.len(), 0);
        })
    });

    #[cfg(not(target_os = "linux"))]
    group.bench_function("riotgames::games", |b| {
        b.iter(|| {
            let games = game_scanner::riotgames::games().unwrap();
            assert_ne!(games.len(), 0);
        })
    });

    #[cfg(not(target_os = "linux"))]
    group.bench_function("steam::games", |b| {
        b.iter(|| {
            let games = game_scanner::steam::games().unwrap();
            assert_ne!(games.len(), 0);
        })
    });

    #[cfg(target_os = "windows")]
    group.bench_function("ubisoft::games", |b| {
        b.iter(|| {
            let games = game_scanner::ubisoft::games().unwrap();
            assert_ne!(games.len(), 0);
        })
    });

    group.finish();
}

criterion_group!(benches, list);
criterion_main!(benches);

File: node/examples/list.js
const {performance, PerformanceObserver} = require('perf_hooks');

const perfObserver = new PerformanceObserver((items) => {
    console.log('\nPerformance status:')
    items.getEntries().forEach((entry) => console.log(`- ${entry.name} ${entry.duration}ms`))
    console.log();
});

perfObserver.observe({entryTypes: ["measure"], buffered: true})

const game_scanner = require("../lib");

try {
    performance.mark('start');
    const games = game_scanner.amazon.games();
    performance.mark('end');
    console.log(games);
    performance.measure("amazon", "start", "end");
} catch (error) {
    console.error(error);
}

try {
    performance.mark('start');
    const games = game_scanner.blizzard.games();
    performance.mark('end');
    console.log(games);
    performance.measure("blizzard", "start", "end");
} catch (error) {
    console.error(error);
}

try {
    performance.mark('start');
    const games = game_scanner.epicgames.games();
    performance.mark('end');
    console.log(games);
    performance.measure("epicgames", "start", "end");
} catch (error) {
    console.error(error);
}

try {
    performance.mark('start');
    const games = game_scanner.gog.games();
    performance.mark('end');
    console.log(games);
    performance.measure("gog", "start", "end");
} catch (error) {
    console.error(error);
}

try {
    performance.mark('start');
    const games = game_scanner.origin.games();
    performance.mark('end');
    console.log(games);
    performance.measure("origin", "start", "end");
} catch (error) {
    console.error(error);
}

try {
    performance.mark('start');
    const games = game_scanner.riotgames.games();
    performance.mark('end');
    console.log(games);
    performance.measure("riotgames", "start", "end");
} catch (error) {
    console.error(error);
}

try {
    performance.mark('start');
    const games = game_scanner.steam.games();
    performance.mark('end');
    console.log(games);
    performance.measure("steam", "start", "end");
} catch (error) {
    console.error(error);
}

try {
    performance.mark('start');
    const games = game_scanner.ubisoft.games();
    performance.mark('end');
    console.log(games);
    performance.measure("ubisoft", "start", "end");
} catch (error) {
    console.error(error);
}
